pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -202,22 +205,64 @@ func (f *ring0Factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, er
 }
 
 func (f *ring0Factory) KubernetesClientSet() (*kubernetes.Clientset, error) {
-	return f.clientCache.KubernetesClientSetForVersion(nil)
```

Comment:
agree we don't need a per-version cache, but it still seems useful to avoid constructing the same client over and over